### PR TITLE
Fix GPU usage with PyTorch / TF

### PR DESCRIPTION
**Pull request by davidslater**

### CODE REVIEW

**Comment by davidslater** [comment]
This ran properly on CARLA multimodal and UCF101 with 100 samples (the latter peaked at ~16 GB GPU memory use).

File: armory/__init__.py [hunk 1]
```diff
@@ -61,6 +61,33 @@ def get_dynamic_version():
...
+# From: https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth
+try:
+    import tensorflow as tf
+
+    gpus = tf.config.list_physical_devices("GPU")
+    if gpus:
+        # Currently, memory growth needs to be the same across GPUs
+        for gpu in gpus:
+            tf.config.experimental.set_memory_growth(gpu, True)
+        logical_gpus = tf.config.list_logical_devices("GPU")
```
Comment (lcadalzo) [inline armory/__init__.py:82]: is there any purpose in obtaining `logical_gpus`? In the TF link it's used for logging purposes, but we deleted that line.

Comment (davidslater) [reply armory/__init__.py:82]: Nope, I'll delete.

**Comment by lcadalzo** [comment]
@davidslater I think there may be a problem caused by this merge, actually. Now torch is being imported in `datasets.py` without any try/except block. This import will fail inside a tf2 container

**Comment by davidslater** [comment]
The tf2 container does have torch installed. But yes, I see your point. Probably good to add back in.

**Pull request merged**